modules/tinymce/src/themes/silver/main/ts/backstage/Backstage.ts

```diff
@@ -41,32 +41,66 @@ export interface UiFactoryBackstage {
   readonly setContextMenuState: (state: boolean) => void;
 }
 
-const init = (lazySink: () => Result<AlloyComponent, string>, editor: Editor, lazyAnchorbar: () => AlloyComponent): UiFactoryBackstage => {
+export interface UiFactoryBackstagePair {
+  readonly popup: UiFactoryBackstage;
+  readonly dialog: UiFactoryBackstage;
+}
+
+const init = (lazySinks: { popup: () => Result<AlloyComponent, string>; dialog: () => Result<AlloyComponent, string> }, editor: Editor, lazyAnchorbar: () => AlloyComponent): UiFactoryBackstagePair => {
   const contextMenuState = Cell(false);
   const toolbar = HeaderBackstage(editor);
-  const backstage: UiFactoryBackstage = {
+
+  const providers: UiFactoryBackstage['shared']['providers'] = {
+    icons: () => editor.ui.registry.getAll().icons,
+    menuItems: () => editor.ui.registry.getAll().menuItems,
+    translate: I18n.translate,
+    isDisabled: () => editor.mode.isReadOnly() || !editor.ui.isEnabled(),
+    getOption: editor.options.get
+  };
+
+  const urlinput = UrlInputBackstage(editor);
+  const styles = initStyleFormatBackstage(editor);
+  const colorinput = ColorInputBackstage(editor);
+  const dialogSettings = DialogBackstage(editor);
+  const isContextMenuOpen = () => contextMenuState.get();
+  const setContextMenuState = (state: boolean) => contextMenuState.set(state);
+
+  const commonBackstage = {
     shared: {
-      providers: {
-        icons: () => editor.ui.registry.getAll().icons,
-        menuItems: () => editor.ui.registry.getAll().menuItems,
-        translate: I18n.translate,
-        isDisabled: () => editor.mode.isReadOnly() || !editor.ui.isEnabled(),
-        getOption: editor.options.get
-      },
-      interpreter: (s) => UiFactory.interpretWithoutForm(s, {}, backstage),
+      providers,
       anchors: Anchors.getAnchors(editor, lazyAnchorbar, toolbar.isPositionedAtTop),
       header: toolbar,
-      getSink: lazySink
     },
-    urlinput: UrlInputBackstage(editor),
-    styles: initStyleFormatBackstage(editor),
-    colorinput: ColorInputBackstage(editor),
-    dialog: DialogBackstage(editor),
-    isContextMenuOpen: () => contextMenuState.get(),
-    setContextMenuState: (state) => contextMenuState.set(state)
+    urlinput,
+    styles,
+    colorinput,
+    dialog: dialogSettings,
+    isContextMenuOpen,
+    setContextMenuState
   };
 
-  return backstage;
+  const popupBackstage: UiFactoryBackstage = {
+    ...commonBackstage,
+    shared: {
+      ...commonBackstage.shared,
+      interpreter: (s) => UiFactory.interpretWithoutForm(s, {}, popupBackstage),
+      getSink: lazySinks.popup
+    }
+  };
+
+  const dialogBackstage: UiFactoryBackstage = {
+    ...commonBackstage,
+    shared: {
+      ...commonBackstage.shared,
+      interpreter: (s) => UiFactory.interpretWithoutForm(s, {}, popupBackstage),
```

Comment:
Is this right? I believe this should be using `dialogBackstage`. If not it probably needs a TODO to change later.

Reply:
Definitely wrong. Wasn't picked up in tests because this would only surface when we tried to create a UI control that cared about the backstage sink ... so I might add a test with a URLInput and check that inline dialogs and modal dialogs are putting the dropdown menu in the right place

- [ ] fix and add test that fails because this was wrong.

Reply:
Ah, the tests use an alternate test backstage, so of course they aren't testing this!

Reply:
Added tests that check these details.